riscv-c-api.md

```diff
@@ -227,14 +226,125 @@ TYPE ::= Optional type postfix.
+| `uint64_t __riscv_aes64es(uint64 rs1, uint64_t rs2);`                   | `aes32es`     | Zkne (RV64)       | |
+| `uint64_t __riscv_aes64esm(uint64 rs1, uint64_t rs2);`                  | `aes32esm`    | Zkne (RV64)       | |
+| `uint32_t __riscv_sha256sig0(uint32_t rs1);`                            | `sha256sig0`  | Zknh              | |
+| `uint32_t __riscv_sha256sig1(uint32_t rs1);`                            | `sha256sig1`  | Zknh              | |
+| `uint32_t __riscv_sha256sum0(uint32_t rs1);`                            | `sha256sum0`  | Zknh              | |
+| `uint32_t __riscv_sha256sum1(uint32_t rs1);`                            | `sha256sum1`  | Zknh              | |
+| `uint32_t __riscv_sha512sig0h(uint32_t rs1, uint32_t rs2);`             | `sha512sig0h` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sig0l(uint32_t rs1, uint32_t rs2);`             | `sha512sig0l` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sig1h(uint32_t rs1, uint32_t rs2);`             | `sha512sig1h` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sig1l(uint32_t rs1, uint32_t rs2);`             | `sha512sig1l` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sum0h(uint32_t rs1, uint32_t rs2);`             | `sha512sum0h` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sum0l(uint32_t rs1, uint32_t rs2);`             | `sha512sum0l` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sum1h(uint32_t rs1, uint32_t rs2);`             | `sha512sum1h` | Zknh (RV32)       | |
+| `uint32_t __riscv_sha512sum1l(uint32_t rs1, uint32_t rs2);`             | `sha512sum1l` | Zknh (RV32)       | |
+| `uint64_t __riscv_sha512sig0(uint64_t rs1);`                            | `sha512sig0`  | Zknh (RV64)       | |
+| `uint64_t __riscv_sha512sig1(uint64_t rs1);`                            | `sha512sig1`  | Zknh (RV64)       | |
+| `uint64_t __riscv_sha512sum0(uint64_t rs1);`                            | `sha512sum0`  | Zknh (RV64)       | |
+| `uint64_t __riscv_sha512sum1(uint64_t rs1);`                            | `sha512sum1`  | Zknh (RV64)       | |
+| `uint32_t __riscv_sm3p0(uint32_t rs1);`                                 | `sm3p0`       | Zksh              | |
+| `uint32_t __riscv_sm3p1(uint32_t rs1);`                                 | `sm3p1`       | Zksh              | |
+| `uint32_t __riscv_sm4ed(uint32_t rs1, uint32_t rs2, const int bs);`     | `sm4ed`       | Zksed             | `bs`=[0..3] |
+| `uint32_t __riscv_sm4ks(uint32_t rs1, uint32_t rs2, const int bs);`     | `sm4ks`       | Zksed             | `bs`=[0..3] |
+
+### Cryptography Intrinsics Implementation Guarantees
+
+The `riscv_crypto.h` can implement the intrinsics in many ways
+(early implementations used inline assembler). Builtin mapping is a
+compiler and system specific issue.
+
+Due to the data-independent latency ("constant time") assertions of
+the `Zkt` extension, the header file or the compiler can't use table
+lookups, conditional branching, etc., when implementing crypto intrinsics.
```

Comment:
Is this actually a useful guarantee? The compiler can do whatever it likes around the intrinsic. If you need data-independent timing then you need to write the entire thing in assembly...